crates/core/src/format.rs

```diff
@@ -188,6 +187,23 @@ impl Format {
         }
     }
 
+    /// Gets a STAC value from an object store.
+    #[cfg(feature = "object-store")]
+    pub async fn get_store<T>(
+        &self,
+        object_store: std::sync::Arc<dyn object_store::ObjectStore>,
```

Comment:
All these fully-qualified paths make my eyes hurt 😆

Reply:
FWIW, similar to https://github.com/apache/arrow-rs/issues/6989#issuecomment-2599842842, you might want an abstraction that implements just the specific elements of `ObjectStore` you need, so that other backends like HTTP can be easily supported as well

Reply:
Oh fair. I also use fully qualified paths inside optional dependency blocks.